dspace-server-webapp/src/main/java/org/dspace/app/rest/ItemAddBundleController.java

```diff
@@ -111,4 +130,61 @@ public ResponseEntity<RepresentationModel<?>> addBundleToItem(@PathVariable UUID
         return ControllerUtils.toResponseEntity(HttpStatus.CREATED, new HttpHeaders(), bundleResource);
     }
 
+    /**
+     * Method to update the license for all bitstreams in the bundle of an item with the given UUID in the URL.
+     * This will remove the old license and attach the new license to all bitstreams in the bundle.
+     * The license is identified by the licenseID parameter in the request. If the licenseID is not found (-1), the old
+     * license will be detached, but the new license will not be attached.
+     * @param uuid
+     * @param licenseId
+     * @param request
+     * @param response
+     * @return
+     * @throws SQLException
+     * @throws AuthorizeException
+     */
+    @RequestMapping(method = RequestMethod.PUT)
+    @PreAuthorize("hasPermission(#uuid, 'ITEM', 'ADD')")
+    public ItemRest updateLicenseForBundle(@PathVariable UUID uuid,
+                                                                  @RequestParam("licenseID") Integer licenseId,
+                                                                  HttpServletRequest request,
+                                                                  HttpServletResponse response)
+            throws SQLException, AuthorizeException {
+        Context context = ContextUtil.obtainContext(request);
+        if (Objects.isNull(context)) {
+            throw new BadRequestException("No context found for current user");
+        }
+        Item item = itemService.find(context, uuid);
+
+        if (item == null) {
+            throw new ResourceNotFoundException("Could not find item with id " + uuid);
+        }
+
+        ClarinLicense clarinLicense = clarinLicenseService.find(context, licenseId);
+        if (Objects.isNull(clarinLicense)) {
+            log.warn("Cannot find clarin license with id: " + licenseId + ". The old license will be detached, " +
+                    "but the new one will not be attached.");
+        }
+        List<Bundle> bundles = item.getBundles(Constants.CONTENT_BUNDLE_NAME);
+        for (Bundle clarinBundle : bundles) {
+            List<Bitstream> bitstreamList = clarinBundle.getBitstreams();
+            for (Bitstream bundleBitstream : bitstreamList) {
+                // in case bitstream ID exists in license table for some reason .. just remove it
+                this.clarinLicenseResourceMappingService.detachLicenses(context, bundleBitstream);
+                if (Objects.nonNull(clarinLicense)) {
+                    // add the license to bitstream
+                    this.clarinLicenseResourceMappingService.attachLicense(context, clarinLicense, bundleBitstream);
+                }
+            }
+        }
+        clarinLicenseService.clearLicenseMetadataFromItem(context, item);
```

Comment:
clear+add = update?

Reply:
Yes, clear old metadata and add a new values - it is common in DSpace